docs/introduction.org

```diff
@@ -0,0 +1,106 @@
+* Introduction
+
+Nowadays, multiple languages have emerged to compete with the well-established C language.
+In fact, C is widely used in many places, learned by a lot of engineers, and used for many purposes.
+C is a powerful tool, /but with great power comes great responsibility/, in fact, memory leaks can easily occur by mistake.
+
+That's why new languages come to find a better way to handle errors and memory.
+The Rust language relies strongly on its borrow checker to avoid safe memory but it comes at a cost: increased code complexity
+
+In another case, the Zig language has a purpose to replace the C language.
+Zig wants to simplify the programming development experience with no hidden control flow and no macro to avoid complex mistakes.
+In comparison with the Rust language, Zig uses explicit memory allocation to prevent memory leaks.
+
+In the sections below, certain aspects of this programming language will be explored in greater depth.
+
+** What is not this documentation
+This is *not* a beginner guide for Zig, this documentation wants to go in-depth in certain parts of the language.
+
+To understand what is described in this documentation, you should first read up on the basics of the Zig language, in particular, its guide, which contains a wealth of information.
+- [[https://zig.guide][Zig guide]]
+- [[https://ziglang.org/documentation/0.11.0][Zig documentation]]
+
+** How to use this documentation
+
+This documentation is built to guide to user through different examples.
+Those examples have been created to use literate programming, in fact, the documentation is built on Emacs with the org mode.
+You can follow the examples and execute it directly in Emacs.
+
+For that, you need to:
+1. Install the configuration created for that (see the section below)
+2. When all is configured, in Emacs you can select the example that you want to execute and use the command ~C-c C-c~.
+   The output will be created or updated below the source block. If you want more documentation, you can check the [[https://orgmode.org/worg/org-contrib/babel/intro.html][Org babel documentation]]
+
+You can follow the examples with the Zig version 0.12.
```

Comment:
I would maybe add that since zig is a constant moving language we MAY have had to use other versions at some point in this guide, but if that is the case we will precise it

Reply:
Fair, I add this